File: BaseballApi/Models/Game.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Security.Permissions;
using Microsoft.EntityFrameworkCore;

namespace BaseballApi.Models;

[Index(nameof(ExternalId), IsUnique = true)]
public class Game
{
    public long Id { get; set; }
    public Guid ExternalId { get; set; }
    public required string Name { get; set; }
    public DateOnly Date { get; set; }
    public GameType? GameType { get; set; }
    public required Team Home { get; set; }
    public required string HomeTeamName { get; set; }
    public required Team Away { get; set; }
    public required string AwayTeamName { get; set; }
    public DateTimeOffset? ScheduledTime { get; set; }
    public DateTimeOffset? StartTime { get; set; }
    public DateTimeOffset? EndTime { get; set; }
    public Park? Location { get; set; }
    public required ICollection<BoxScore> BoxScores { get; set; }
    public long? HomeBoxScoreId { get; set; }
    [ForeignKey(nameof(HomeBoxScoreId))]
    public BoxScore? HomeBoxScore { get; set; }
    public long? AwayBoxScoreId { get; set; }
    [ForeignKey(nameof(AwayBoxScoreId))]
    public BoxScore? AwayBoxScore { get; set; }
    public int? HomeScore { get; set; }
    public int? AwayScore { get; set; }
    public Team? WinningTeam { get; set; }
    public Team? LosingTeam { get; set; }
    public Player? WinningPitcher { get; set; }
    public Player? LosingPitcher { get; set; }
    public Player? SavingPitcher { get; set; }
}

File: iScoreImport/iScoreImport/Models/Game.swift
//
//  Game.swift
//  iScoreImport
//
//  Created by Patrick Eschenfeldt on 6/29/24.
//

import Foundation

struct Game : Codable {
    let Name: String
    let ExternalId: UUID?
    let Date: Date
    let HomeTeam: Team
    let AwayTeam: Team
    let ScheduledTime: Date?
    let StartTime: Date?
    let EndTime: Date?
    let Location: Park?
    var HomeScore: Int?
    var AwayScore: Int?
    let HomeTeamName: String
    let AwayTeamName: String
    var WinningTeam: Team?
    var LosingTeam: Team?
    let WinningPitcher: Player?
    let LosingPitcher: Player?
    let SavingPitcher: Player?
    var HomeBoxScore: BoxScore?
    var AwayBoxScore: BoxScore?
    
    mutating func assignWinner() {
        guard let homeScore = self.HomeScore, let awayScore = self.AwayScore else { return }
        if homeScore > awayScore {
            self.WinningTeam = self.HomeTeam
            self.LosingTeam = self.AwayTeam
        } else if awayScore > homeScore {
            self.WinningTeam = self.AwayTeam
            self.LosingTeam = self.HomeTeam
        }
    }
}
